open_webui/pipelines/main.py

```diff
@@ -231,3 +210,41 @@ def stream_content():
 @app.get("/")
 async def get_status():
     return {"status": True}
+
+
+def callback(
+    pipelines: Optional[Path] = None
+):
+    for loaded_module in load_modules_from_directory(pipelines):
+        # Do something with the loaded module
+        logging.info("Loaded:", loaded_module.__name__)
+
+        pipeline = loaded_module.Pipeline()
+
+        PIPELINES[loaded_module.__name__] = {
+            "module": pipeline,
+            "id": pipeline.id if hasattr(pipeline, "id") else loaded_module.__name__,
+            "name": pipeline.name if hasattr(pipeline, "name") else loaded_module.__name__,
+        }
+
+cli = typer.Typer(callback=callback)
```

Comment:
Ideally we'd want to keep typer separate from the main.py like our main open webui repo.

Reply:
> Ideally we'd want to keep typer separate from the main.py like our main open webui repo.

OK, I will put typer in another file